fe/fe-core/src/main/java/com/starrocks/common/Config.java

```diff
@@ -1115,6 +1115,17 @@ public class Config extends ConfigBase {
     @ConfField(mutable = true)
     public static boolean enable_backup_materialized_view = true;
 
+    /**
+     * To avoid too many related materialized view causing too much fe memory and decreasing performance, set N
+     * to determine which strategy you choose:
+     *  N <0      : always use non lock optimization and no copy related materialized views which
+     *      may cause metadata concurrency problem but can reduce many lock conflict time and metadata memory-copy consume.
+     *  N = 0    : always not use non lock optimization
+     *  N > 0    : use non lock optimization when related mvs's num <= N, otherwise don't use non lock optimization
+     */
+    @ConfField(mutable = true)
+    public static int non_lock_optimization_max_related_materialized_view_num = 5;
```

Comment:
Put this config item into GlobalVariables?    Config command is hard to remember and write.

use a clear name? for an example
```
skip_whole_phase_lock_mv_limit
```